src/main/ml-modules/root/com.marklogic.smart-mastering/matcher-impl/matcher-impl.xqy

```diff
@@ -111,8 +111,6 @@ declare function match-impl:find-document-matches-by-options(
       match-impl:lock-on-search($serialized-match-query/cts:and-query/cts:or-query)
     else ()
   let $matches :=
-    match-impl:drop-redundant(
```

Comment:
Removing drop-redundant function. It prevented iterative mastering of data and the sm-match-and-merge endpoint that now handles URIs in batches solves the problem previously seen.